server/accounts.go

```diff
@@ -925,17 +939,67 @@ func buildInternalNkeyUser(uc *jwt.UserClaims, acc *Account) *NkeyUser {
 
 // AccountResolver interface. This is to fetch Account JWTs by public nkeys
 type AccountResolver interface {
-	Fetch(pub string) (string, error)
+	Fetch(name string) (string, error)
+	Store(name, jwt string) error
 }
 
 // Mostly for testing.
 type MemAccResolver struct {
-	sync.Map
+	sm sync.Map
 }
 
-func (m *MemAccResolver) Fetch(pub string) (string, error) {
-	if j, ok := m.Load(pub); ok {
+func (m *MemAccResolver) Fetch(name string) (string, error) {
+	if j, ok := m.sm.Load(name); ok {
 		return j.(string), nil
 	}
-	return "", ErrMissingAccount
+	return _EMPTY_, ErrMissingAccount
+}
+
+func (m *MemAccResolver) Store(name, jwt string) error {
+	m.sm.Store(name, jwt)
+	return nil
+}
+
+//
+type URLAccResolver struct {
+	url string
+	c   *http.Client
+}
+
+// NewURLAccResolver returns a new resolver for the given base URL.
+func NewURLAccResolver(url string) (*URLAccResolver, error) {
+	if !strings.HasSuffix(url, "/") {
+		url += "/"
+	}
+	// Do basic test to see if anyone is home.
+	// FIXME(dlc) - Make timeout configurable post MVP.
+	ur := &URLAccResolver{
+		url: url,
+		c:   &http.Client{Timeout: 2 * time.Second},
+	}
+	if _, err := ur.Fetch(""); err != nil {
+		return nil, err
+	}
+	return ur, nil
+}
+
+// Fetch will fetch the account jwt claims from the base url, appending the
+// account name onto the end.
+func (ur *URLAccResolver) Fetch(name string) (string, error) {
+	url := ur.url + name
+	resp, err := ur.c.Get(url)
```

Comment:
Should we have a timeout and maybe block redirects?

Reply:
We have timeouts already. They are fixed atm but we do set them. Not sure about redirects though. Will take a look.

Reply:
I think timeout being set covers us for wonky redirects.